src/components/picker/index.tsx

```diff
@@ -267,14 +252,14 @@ const Picker = React.forwardRef((props: PickerProps, ref) => {
         /* @ts-expect-error */
         <ExpandableOverlay
           ref={pickerExpandable}
-          useDialog={useDialog || useWheelPicker}
-          modalProps={modalProps}
-          dialogProps={customPickerProps?.dialogProps || DIALOG_PROPS}
+          useDialog={type.dialog || type.wheelPicker}
+          modalProps={componentProps.pickerModalProps}
           expandableContent={expandableModalContent}
-          renderCustomOverlay={renderCustomModal ? _renderCustomModal : undefined}
+          renderCustomOverlay={type.custom ? _renderCustomModal : undefined}
           onPress={onPress}
           testID={testID}
           {...customPickerProps}
+          dialogProps={componentProps.dialogProps}
```

Comment:
The dialogProps are overriding the customPickerProps. Question: is the customPickerProps prop should be applied only when pickerType=custom? What is this prop for? The description is not clear